app/assets/javascripts/pretty-text/addon/upload-short-url.js

```diff
@@ -53,6 +61,10 @@ function retrieveCachedUrl(upload, siteSettings, dataAttribute, callback) {
     upload.removeAttribute(`data-${dataAttribute}`);
     if (url !== MISSING) {
       callback(url);
+    } else if (opts && opts.removeMissing) {
+      const stub = document.createElement("p");
```

Comment:
Does that work on the server side? (eg. when cooking a post on the server via [`pretty-text`](https://github.com/discourse/discourse/blob/0a1d66929039699388a9b1a583bff3702f81d419/lib/pretty_text.rb))